src/main/java/de/hpi/bpt/argos/core/ArgosImpl.java

```diff
@@ -1,143 +1,70 @@
-import de.hpi.bpt.argos.eventHandling.EventPlatformRestEndpoint;
-import de.hpi.bpt.argos.eventHandling.EventPlatformRestEndpointImpl;
-import de.hpi.bpt.argos.persistence.database.PersistenceEntityManager;
-import de.hpi.bpt.argos.persistence.database.PersistenceEntityManagerImpl;
-import de.hpi.bpt.argos.persistence.model.event.statusUpdate.StatusUpdateEventTypeImpl;
-import de.hpi.bpt.argos.persistence.model.event.type.EventType;
-import de.hpi.bpt.argos.persistence.model.parsing.BackboneDataParserImpl;
-import de.hpi.bpt.argos.persistence.model.parsing.DataFile;
-import de.hpi.bpt.argos.properties.PropertyEditor;
-import de.hpi.bpt.argos.properties.PropertyEditorImpl;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import de.hpi.bpt.argos.common.EventProcessingPlatformUpdaterImpl;
+import de.hpi.bpt.argos.common.RestEndpoint;
+import de.hpi.bpt.argos.eventProcessing.EventReceiverImpl;
+import de.hpi.bpt.argos.notifications.ClientUpdateServiceImpl;
+import de.hpi.bpt.argos.storage.PersistenceAdapterImpl;
 import spark.Service;
 
-import java.io.File;
-import java.nio.charset.StandardCharsets;
-import java.nio.file.Files;
-import java.nio.file.Paths;
-
-import static spark.Service.ignite;
+import java.util.HashSet;
+import java.util.Set;
 
 /**
  * {@inheritDoc}
  * This is the implementation.
  */
 public class ArgosImpl implements Argos {
-	private static final Logger logger = LoggerFactory.getLogger(ArgosImpl.class);
-
-	protected static final JsonParser jsonParser = new JsonParser();
-	protected Service sparkService;
-	protected PersistenceEntityManager entityManager;
-
-	protected CustomerRestEndpoint customerRestEndpoint;
-	protected EventPlatformRestEndpoint eventPlatformRestEndpoint;
+	private Service sparkService;
 
-    /**
-     * {@inheritDoc}
-     */
+	/**
+	 * {@inheritDoc}
+	 */
 	@Override
-	public void run(int port, int numberOfThreads) {
-
-		sparkService = startServer(port, numberOfThreads);
-
-		entityManager = new PersistenceEntityManagerImpl();
-		if (!entityManager.setup()) {
-			shutdown();
+	public void start() {
+		sparkService = Service.ignite()
+				.port(Argos.getPort())
+				.threadPool(Argos.getThreads())
+				.staticFileLocation(Argos.getPublicFiles());
+
+		if (!PersistenceAdapterImpl.getInstance().establishConnection()) {
+			stop();
 			return;
 		}
 
-		loadDefaultEventTypes();
-		loadBackboneData();
+		// TODO: parse default event types and entity data
+
+		// keep this order, since web sockets should be registered before any web routes get registered
+		(new ClientUpdateServiceImpl()).setup(sparkService);
+		EventProcessingPlatformUpdaterImpl.getInstance().setup();
 
-		ResponseFactory responseFactory = new ResponseFactoryImpl();
-		customerRestEndpoint = new CustomerRestEndpointImpl();
-		eventPlatformRestEndpoint = new EventPlatformRestEndpointImpl();
+		Set<RestEndpoint> restEndpoints = new HashSet<>();
+		restEndpoints.add(new EventReceiverImpl());
+		// TODO: add more restEndpoints here
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS1135)